src/NuGet.Core/NuGet.Packaging/Core/FrameworkNameValidatorUtility.cs

```diff
@@ -34,6 +34,60 @@ internal static bool IsValidFrameworkName(string path)
             return fx != null && fx.Framework != NuGetFramework.UnsupportedFramework.Framework;
         }
 
+        internal static bool IsDottedFrameworkVersion(string path)
+        {
+            foreach (string knownFolder in PackagingConstants.Folders.Known)
+            {
+                string folderPrefix = knownFolder + System.IO.Path.DirectorySeparatorChar;
+                if (knownFolder == "contentFiles")
+                {
+                    folderPrefix += "any" + System.IO.Path.DirectorySeparatorChar;
+                }
+                if (path.Length > folderPrefix.Length &&
+                    path.StartsWith(folderPrefix, StringComparison.OrdinalIgnoreCase))
+                {
+                    string frameworkPart = path.Substring(folderPrefix.Length);
+
+                    try
+                    {
+                        string effectivePath = null;
+                        NuGetFramework framework = FrameworkNameUtility.ParseNuGetFrameworkFolderName(
+                            frameworkPart,
+                            strictParsing: knownFolder == PackagingConstants.Folders.Lib,
```

Comment:
A question here: What's the reasoning behind strict parsing for the lib folder only? Are we following a precedent from somewhere? 
If so, can you point me to that, cause I'd love to understand that.

Reply:
I just copy-pasted this from elsewhere. I'll remove it.